### PR TITLE
CB-6579 - Azure single RG: Env service REST changes

### PR DESCRIPTION
**Pull request by gergopapi2**

### CODE REVIEW

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceGroupRequest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel(value = "AzureResourceGroupV1Parameters")
+public class AzureResourceGroupRequest {
+
+    @ApiModelProperty(EnvironmentModelDescription.EXISTING_RESOURCE_GROUP_NAME)
+    private String name;
+
+    @ApiModelProperty(EnvironmentModelDescription.USE_SINGLE_RESOURCE_GROUP)
+    private Boolean single;
```
Comment (doktoric) [inline environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceGroupRequest.java:15]: I guess the default value should be true

Comment (doktoric) [reply environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceGroupRequest.java:15]: can we rename something else ?

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/response/DetailedEnvironmentResponse.java [hunk 2]
```diff
@@ -87,6 +87,8 @@ public void setProxyConfig(ProxyResponse proxyConfig) {
 
         private AwsEnvironmentParameters aws;
 
+        private AzureEnvironmentParametersResponse azure;
```
Comment (doktoric) [inline environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/response/DetailedEnvironmentResponse.java:90]: AzureEnvironmentParametersResponse -> AzureEnvironmentParameters

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/response/EnvironmentBaseResponse.java [hunk 3]
```diff
@@ -77,6 +77,9 @@
     @ApiModelProperty(EnvironmentModelDescription.AWS_PARAMETERS)
     private AwsEnvironmentParameters aws;
 
+    @ApiModelProperty(EnvironmentModelDescription.AZURE_PARAMETERS)
+    private AzureEnvironmentParametersResponse azure;
```
Comment (doktoric) [inline environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/response/EnvironmentBaseResponse.java:81]: AzureEnvironmentParametersResponse -> AzureEnvironmentParameters

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/response/SimpleEnvironmentResponse.java [hunk 4]
```diff
@@ -74,6 +74,8 @@ public static Builder builder() {
 
         private AwsEnvironmentParameters aws;
 
+        private AzureEnvironmentParametersResponse azure;
```
Comment (doktoric) [inline environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/response/SimpleEnvironmentResponse.java:77]: AzureEnvironmentParametersResponse -> AzureEnvironmentParameters

File: environment/src/main/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentApiConverter.java [hunk 5]
```diff
@@ -105,9 +109,11 @@ public EnvironmentCreationDto initCreationDto(EnvironmentRequest request) {
                         .withCloudStorageValidation(request.getCloudStorageValidation())
                         .withTunnel(tunnelConverter.convert(request.getTunnel()))
                         .build())
-                .withParameters(awsParamsToParametersDto(
+                .withParameters(paramsToParametersDto(
```
Comment (doktoric) [inline environment/src/main/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentApiConverter.java:112]: can we create a separated method for azure ?

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/response/DetailedEnvironmentResponse.java (hunk 2 above)
Comment (gergopapi2) [reply environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/response/DetailedEnvironmentResponse.java:90]: Ok, and can AzureEnvironmentParametersRequest stay the same?

File: environment/src/main/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentResponseConverter.java [hunk 6]
```diff
@@ -81,6 +86,7 @@ public DetailedEnvironmentResponse dtoToDetailedResponse(EnvironmentDto environm
                 .withCloudStorageValidation(environmentDto.getExperimentalFeatures().getCloudStorageValidation())
                 .withAdminGroupName(environmentDto.getAdminGroupName())
                 .withAws(getIfNotNull(environmentDto.getParameters(), this::awsEnvParamsToAwsEnvironmentParams))
+                .withAzure(getIfNotNull(environmentDto.getParameters(), this::azureEnvParamsToAwsEnvironmentParams))
```
Comment (doktoric) [inline environment/src/main/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentResponseConverter.java:89]: azureEnvParamsToAwsEnvironmentParams -> azureEnvParamsToAzureEnvironmentParams

File: environment/src/test/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentApiConverterTest.java [hunk 7]
```diff
@@ -185,6 +188,10 @@ private void assertExperimentalFeatures(EnvironmentRequest request, Experimental
     private void assertParameters(EnvironmentRequest request, ParametersDto actual) {
         assertEquals(request.getAws().getS3guard().getDynamoDbTableName(), actual.getAwsParametersDto().getS3GuardTableName());
```
Comment (doktoric) [inline environment/src/test/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentApiConverterTest.java:189]: please create a separated test only for Azure

File: environment/src/test/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentResponseConverterTest.java [hunk 8]
```diff
@@ -175,6 +180,7 @@ void testDtoToSimpleResponse() {
         assertEquals(telemetryResponse, actual.getTelemetry());
         assertEquals(compactRegionResponse, actual.getRegions());
         assertEquals(environment.getParameters().getAwsParametersDto().getS3GuardTableName(), actual.getAws().getS3guard().getDynamoDbTableName());
+        assertAzureParameters(environment.getParameters().getAzureParametersDto(), actual.getAzure());
```
Comment (doktoric) [inline environment/src/test/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentResponseConverterTest.java:183]: Please create a pull request only for Azure

**Comment by doktoric** [comment]
I guess if Azure and Aws parameters defined then it is a badrequest right ?

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceGroupRequest.java (hunk 1 above)
Comment (gergopapi2) [reply environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceGroupRequest.java:15]: > I guess the default value should be true

No, for now false. When the feature is ready then we can set it to true.

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/EnvironmentRequest.java [hunk 9]
```diff
@@ -95,6 +96,10 @@
     @ApiModelProperty(EnvironmentModelDescription.AWS_PARAMETERS)
     private AwsEnvironmentParameters aws;
 
+    @Valid
+    @ApiModelProperty(EnvironmentModelDescription.AWS_PARAMETERS)
+    private AzureEnvironmentParametersRequest azure;
```
Comment (doktoric) [inline environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/EnvironmentRequest.java:101]: AzureEnvironmentParametersRequest -> AzureEnvironmentParameters

File: environment/src/test/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentResponseConverterTest.java (hunk 8 above)
Comment (gergopapi2) [reply environment/src/test/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentResponseConverterTest.java:183]: What do you mean by that?

Comment (gergopapi2) [reply environment/src/test/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentResponseConverterTest.java:183]: Clarified: separate test method for azure

**Pull request merged**